Framework/Kernel/inc/MantidKernel/LambdaValidator.h

```diff
@@ -0,0 +1,32 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidKernel/TypedValidator.h"
+#include <functional>
+
+namespace Mantid::Kernel {
+
+template <typename ParamType> class DLLExport LambdaValidator : public TypedValidator<ParamType> {
+  using ValidatorFunction = std::function<std::string(ParamType)>;
+
+public:
+  LambdaValidator(const ValidatorFunction &validatorFunction) : m_validatorFunction(validatorFunction){};
```

Comment:
For consistency with the rest of the validators, could you implement the default constructor that just sets `m_validatorFunction` to be an empty lambda?

Reply:
ok

Reply:
Having had a look at other validators, if they are unset they tend to just return "" rather than an error, so probably do that here for consistency?

Reply:
I just thought of it that the default behaviour should be invalid by default so that if you use the default constructor the developer should remember to add a function

Reply:
Yeah I thought that initially as well, but checked the behaviour of the other validators. I'm happy with either way to be honest.